tests/chainer_tests/links_tests/connection_tests/test_deconvolution_nd.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import gradient_check
+from chainer import initializers
+from chainer.links import deconvolution_nd
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+from chainer.testing import parameterize
+from chainer.utils import conv
+
+
+@parameterize(*testing.product({
+    'dims': [(5, 4, 3), (4, 3), (3,)],
+    'nobias': [True, False],
+    'dtype': [numpy.float32],   # TODO(takagi) other dtypes.
+    'use_cudnn': [True, False],
+    'use_outsize': [True, False],
+}))
+class TestDeconvolutionND(unittest.TestCase):
+
+    def setUp(self):
+        N = 2
+        in_channels = 3
+        out_channels = 2
+        ndim = len(self.dims)
+        ksize = (3,) * ndim
+        stride = (2,) * ndim
+        pad = (1,) * ndim
+
+        outs = tuple(
+            conv.get_deconv_outsize(d, k, s, p)
+            for (d, k, s, p) in zip(self.dims, ksize, stride, pad))
```

Comment:
Sure, I will add tests with output size with values `get_deconv_outsize` does not give, thanks for the clear explanation.